integration-tests/pkg/executor/executor_docker_api.go

```diff
@@ -130,6 +130,28 @@ func (d *dockerAPIExecutor) StartContainer(startConfig config.ContainerStartConf
 		return "", errors.Wrapf(err, "start %s", startConfig.Name)
 	}
 
+	tickSeconds := time.Second
+	timeout := 30 * time.Second
+
+	tick := time.Tick(tickSeconds)
+	timer := time.After(timeout)
+
+	for {
+		select {
+		case <-tick:
+			inspect, err := d.client.ContainerExecInspect(ctx, resp.ID)
```

Comment:
We might want to directly check if there was an error here:
```suggestion
			inspect, err := d.client.ContainerExecInspect(ctx, resp.ID)
			if err != nil {
				log.Info("Failed to inspect %s: %s", startConfig.Name, err)
				continue
			}
```

Then we can remove it from `err` from the log message on line 148, since it's currently causing some noise:
```
2024/10/21 15:41:38 INFO: Wait for container container-stats to start, %!w(errdefs.errNotFound=***0xc000616a08***)
2024/10/21 15:41:39 INFO: Wait for container container-stats to start, %!w(errdefs.errNotFound=***0xc000616b58***)
2024/10/21 15:41:40 INFO: Wait for container container-stats to start, %!w(errdefs.errNotFound=***0xc000616d08***)
2024/10/21 15:41:41 INFO: Wait for container container-stats to start, %!w(errdefs.errNotFound=***0xc000616fa8***)
```